### DISCUSSION TITLE
Server specification (other servers)

### DISCUSSION THREAD

**Discussion by shulcsm** [trpc/trpc#4174] [General]
Assuming i want to implement server side in other language is there anything besides this: https://trpc.io/docs/rpc ?
It seems like only implementation detail is the error codes.
So would any JSON-RPC 2.0 complaint server work?

**Comment by KATT** [comment]
That's the goto doc yes, it should be JSON-RPC compliant

**Reply by crabmusket** [reply]
The batching doesn't seem compatible to me? JSON-RPC requires that each request and response in a batch have IDs to match them up.

Nor does trpc seem to use the `method` key, putting methods in the URL instead?

**Reply by KATT** [reply]
Our http layer doesn't really use json-rpc in that way in the calls but the wss layer does. I'm curious, what is the problem you're trying to solve?

**Reply by crabmusket** [reply]
Me personally, I'm just investigating tRPC for potential use and came across this discussion. I've implemented JSON-RPC before and was interested to look into what tRPC was using under the hood. I came across this discussion and it seemed slightly off to me!

Not that this seems very critical, but I'm curious: could a custom HTTP link implement "fully compliant" JSON-RPC behaviour so that the client works with a JSON-RPC server in a different language?

I get that's voiding a lot of the benefits of using tRPC, I just like the idea of interop standards!
